### PR TITLE
Add precompile for pallet metadata extender

### PR DESCRIPTION
**Pull request by asiniscalchi**

### CODE REVIEW

**Review by asiniscalchi — COMMENTED** [review]
some comments

File: ownership-chain/pallets/asset-metadata-extender/src/lib.rs [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 pub use pallet::*;
-use sp_runtime::{traits::One, ArithmeticError, DispatchResult};
-use traits::AssetMetadataExtender;
-use types::*;
+use sp_core::H160;
+use sp_runtime::{
+	traits::{Convert, One},
+	ArithmeticError, DispatchResult,
+};
+pub use traits::AssetMetadataExtender as AssetMetadataExtenderT;
```
[the remark this answers is not in the document]
Comment (dastanfv) [reply ownership-chain/pallets/asset-metadata-extender/src/lib.rs:15]: I think this is Carla's code. but in general, this is done when you have a type/struct which has a similar name to a trait. Appending `T` helps to disambiguate and also fixes the compile error

File: ownership-chain/pallets/asset-metadata-extender/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+		#[block]
+		{
+			AssetMetadataExtender::<T>::create_token_uri_extension(
+				claimer.clone(),
+				vec![1u8; t as usize].try_into().unwrap(),
+				vec![1u8; u as usize].try_into().unwrap(),
+			)
+			.unwrap();
+		};
```
[the remark this answers is not in the document]
Comment (dastanfv) [reply ownership-chain/pallets/asset-metadata-extender/src/benchmarking.rs:35]: I don't think cloning and vector initiation + conversion to `BoundedVec` would impact the weights significantly. I checked locally after removing clone and moving bounded vec creation out of `#[block]` and there's really no difference at all. Obviously not exactly the same weight but weights are not deterministic, so a very small difference is from that

```
Current:
 > Weight::from_parts(18_329_058, 4545)

with changes:

> Weight::from_parts(18_286_268, 4545)
```

File: ownership-chain/pallets/asset-metadata-extender/src/benchmarking.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+		let ul: UniversalLocationOf<T> = vec![1u8; u as usize].try_into().unwrap();
+
+		assert_eq!(
```
[the remark this answers is not in the document]
Comment (dastanfv) [reply ownership-chain/pallets/asset-metadata-extender/src/benchmarking.rs:39]: yes, they are. it's done in benchmarks also to verify the storage was altered after the call. For example, here in the [`pallet-balances`](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/balances/src/benchmarking.rs#L203)

 in the old version there was a separate `verify {}` block where you would do the checks, but with this new version we can do assertions like these.

**Review by tonimateos — DISMISSED** [review]

**Review by dastanfv — DISMISSED** [review]

**Review by tonimateos — DISMISSED** [review]

**Review by tonimateos — APPROVED** [review]

**Review by ccubu — APPROVED** [review]

**Pull request merged**